services/director-v2/tests/unit/test_modules_osparc_variables.py

```diff
@@ -186,10 +190,12 @@ async def test_resolve_and_substitute_session_variables_in_specs(
         project_id=faker.uuid4(cast_to=None),
         node_id=faker.uuid4(cast_to=None),
         service_run_id=ServiceRunID("some_run_id"),
+        wallet_id=wallet_id,
     )
     print("REPLACED SPECS\n", replaced_specs)
 
     assert OSPARC_IDENTIFIER_PREFIX not in f"{replaced_specs}"
+    assert f"{'' if wallet_id is None else wallet_id}" in f"{replaced_specs}"
```

Comment:
as I said, i think this should either be a general policy of the substitution mechanism (i.e. `None` values get substituted as empty strings ) and/or we have to add this description

Reply:
Note that there is not a unique obvious way and the user could also interpret that:
- The substitution is `OSPARC_VARIABLE_WALLET_ID=null`
- Raises an error (as BTW you suggested when the osparc variable does not exist)
- The substitution is `OSPARC_VARIABLE_WALLET_ID=None`
- The substitution is `OSPARC_VARIABLE_WALLET_ID=`

Reply:
Currently the substations is `OSPARC_VARIABLE_WALLET_ID=None`. Users have to deal with it. The issue is that `None` is a python concept. What about C++ or other languages.

Reply:
Above is the current undocumented behaviour on which other pieces of the code rely upon. I would if we should change it.

Reply:
After a chat with MaG we can leave it as is resolving to `OSPARC_VARIABLE_WALLET_ID=None` in case `wallet_id=None`